vms/platformvm/txs/mempool/mempool.go

```diff
@@ -50,7 +53,10 @@ type Mempool interface {
 	EnableAdding()
 	DisableAdding()
 
-	Add(tx *txs.Tx) error
+	// Add allows inserting a transaction in mempool.
+	// Timestamp is the chain tip time. It's needed to
+	// handle hard forks
+	Add(tx *txs.Tx, timestamp time.Time) error
```

Comment:
Why do we need to handle mandatory upgrades in the mempool?

Reply:
Add AddValidator/Delegator.. transactions satisfies PreDForkStaker and Staker interfaces. We need the chain time to understand whether we should treat these txs as one type or the other. In mempool PreDForkStakers are ordered by start time which does not make sense for post fork StakerTxs.
The chain time helps controlling which behaviour apply.
Once the fork is activated, we will clean this all up and much simplify the mempool

Reply:
I think it might make more sense to revive this PR https://github.com/ava-labs/avalanchego/pull/1536

Reply:
Following fork activation we won't need anymore to sort txs by age, so we'll be able to cleanup the mempool much more aggressively than what is done in https://github.com/ava-labs/avalanchego/pull/1536 (see [this comment](https://github.com/ava-labs/avalanchego/pull/1536#discussion_r1209457134))
For the time being I picked the approach of just passing chainTime because is the one involving the least code changes.